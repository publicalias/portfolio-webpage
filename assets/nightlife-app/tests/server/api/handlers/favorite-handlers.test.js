"use strict";

//local imports

const handlers = require("../../../../scripts/server/api/handlers/favorite-handlers");

const { newFavorite } = require("../../../../schemas");

//global imports

const { newUser } = require("redux/schemas");
const { testMock } = require("redux/tests/meta-tests");
const { initMockAPICall, mongoTests, testAuthFail, testInsert } = require("redux/tests/server-tests");

//utilities

const favoritesCol = () => db.collection("nightlife-app/favorites");

//setup

beforeAll(mongoTests.setup);
beforeEach(mongoTests.reset(favoritesCol));
afterAll(mongoTests.teardown);

//favorite add

describe("favoriteAdd", () => {

  const { favoriteAdd } = handlers;

  const mockAPICall = initMockAPICall(favoriteAdd, "POST");

  const getData = () => ({
    name: "Venue A",
    id: "id-a"
  });

  it("sends 401 if authentication fails", async () => {

    await testAuthFail(mockAPICall, getData());

    expect(await favoritesCol().countDocuments()).toEqual(0);

  });

  it("sends noop if successful", async () => {

    const res = await mockAPICall(
      newUser({
        name: "User A",
        id: "id-a"
      }),
      getData()
    );

    testMock(res.json, [{}]);

    await testInsert(favoritesCol);

  });

});

//favorite remove

describe("favoriteRemove", () => {

  const { favoriteRemove } = handlers;

  const mockAPICall = initMockAPICall(favoriteRemove, "DELETE");

  const getData = () => ({ id: "id-a" });

  beforeEach(() => favoritesCol().insertOne(newFavorite({
    id: "id-a",
    user: { id: "id-b" }
  })));

  it("sends 401 if authentication fails", async () => {

    await testAuthFail(mockAPICall, getData(), [newUser({ id: "id-c" })]);

    expect(await favoritesCol().countDocuments()).toEqual(1);

  });

  it("sends noop if successful", async () => {

    const res = await mockAPICall(newUser({ id: "id-b" }), getData());

    testMock(res.json, [{}]);

    expect(await favoritesCol().countDocuments()).toEqual(0);

  });

});
