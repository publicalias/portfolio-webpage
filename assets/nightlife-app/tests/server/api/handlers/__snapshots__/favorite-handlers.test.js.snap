// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoriteAdd sends object if user is authenticated 1`] = `
Object {
  "_id": "[CENSORED]",
  "id": "[CENSORED]",
  "user": Object {
    "id": "id-a",
    "name": "User A",
  },
  "venue": Object {
    "id": "id-a",
    "name": "Venue A",
  },
}
`;
